tests/pubsub/testgossipmembership.nim

```diff
@@ -146,17 +146,16 @@ suite "GossipSub Topic Membership Tests":
 
   # Test ensuring that the number of subscriptions does not exceed the limit set in the GossipSub parameters
   asyncTest "subscription limit test":
-    let gossipSub = TestGossipSub.init(newStandardSwitch())
-    gossipSub.topicsHigh = 10
+    let topicCount = 15 # Total number of topics to be tested
+    let gossipSubParams = 10 # Subscription limit for the topics
+    let topicNames = toSeq(mapIt(0 .. topicCount - 1, "topic" & $it))
 
-    var conns = newSeq[Connection]()
-    for i in 0 .. gossipSub.topicsHigh + 5:
-      let topic = "topic" & $i
-      # Ensure all topics are properly initialized before subscribing
-      gossipSub.mesh[topic] = initHashSet[PubSubPeer]()
-      gossipSub.topicParams[topic] = TopicParams.init()
-      gossipSub.gossipsub[topic] = initHashSet[PubSubPeer]()
+    # Use setupGossipSub to initialize the GossipSub system with connections
+    let (gossipSub, conns) = setupGossipSub(topicNames, 0) # No peers for now
+
+    gossipSub.topicsHigh = gossipSubParams # Set the subscription limit
 
+    for topic in topicNames:
       if gossipSub.topics.len < gossipSub.topicsHigh:
```

Comment:
Now that I think about this, this doesn't make a lot of sense. The limit shouldn't be handled by the test, not manually by us. This is what the test should be asserting, actually: That regardless of the amount of subscriptions done, it shouldn't exceed the limit.

P.S.: Please, remove all the `PubSub` usages you added as these are `GossipSub` cases.